src/preferences/upgrade.cpp

```diff
@@ -433,9 +434,18 @@ UserSettingsPointer Upgrade::versionUpgrade(const QString& settingsPath) {
         }
     }
 
-    if (configVersion.startsWith("1.12") ||
-        configVersion.startsWith("2.0") ||
-        configVersion.startsWith("2.1.0")) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+    const QStringList versionParts = configVersion.split(".");
+    const QVersionNumber configFileVersion(versionParts[0].toInt(),
```

Comment:
we prefer initializing by assignment, because this reads more natural: 
```suggestion
    const auto configFileVersion = QVersionNumber(versionParts[0].toInt(),
```